Firestore/Source/API/FIRTransaction.mm

```diff
@@ -119,16 +119,19 @@ - (void)getDocument:(FIRDocumentReference *)document
                     HARD_ASSERT(documents.count == 1,
                                 "Mismatch in docs returned from document lookup.");
                     FSTMaybeDocument *internalDoc = documents.firstObject;
-                    if ([internalDoc isKindOfClass:[FSTDeletedDocument class]]) {
+                    if ([internalDoc isKindOfClass:[FSTMaybeDocument class]]) {
```

Comment:
On line 129 you check to see if internalDoc is a FSTDeletedDocument. I'm not sure how rtti works in objc, but since FSTDeletedDoc inherits from MaybeDoc, doesn't the check here make the one below impossible?

If yes, then I think changing the check here to be for FSTDocument might be the solution.

Reply:
Oh, this should check for FSTDocument. Otherwise the cast below will throw. Thanks for catching this!